--- lib/vendor/univ2core/contracts/UniswapV2Pair.sol	2026-08-24 16:41:13
+++ src/patched/UniswapV2PairZeroFee.sol	2026-08-24 12:02:08
@@ -1,14 +1,14 @@
 pragma solidity =0.5.16;
 
-import './interfaces/IUniswapV2Pair.sol';
-import './UniswapV2ERC20.sol';
-import './libraries/Math.sol';
-import './libraries/UQ112x112.sol';
-import './interfaces/IERC20.sol';
-import './interfaces/IUniswapV2Factory.sol';
-import './interfaces/IUniswapV2Callee.sol';
+import 'vendor/univ2core/contracts/interfaces/IUniswapV2Pair.sol';
+import 'vendor/univ2core/contracts/UniswapV2ERC20.sol';
+import 'vendor/univ2core/contracts/libraries/Math.sol';
+import 'vendor/univ2core/contracts/libraries/UQ112x112.sol';
+import 'vendor/univ2core/contracts/interfaces/IERC20.sol';
+import 'vendor/univ2core/contracts/interfaces/IUniswapV2Factory.sol';
+import 'vendor/univ2core/contracts/interfaces/IUniswapV2Callee.sol';
 
-contract UniswapV2Pair is IUniswapV2Pair, UniswapV2ERC20 {
+contract UniswapV2PairZeroFee is IUniswapV2Pair, UniswapV2ERC20 {
     using SafeMath  for uint;
     using UQ112x112 for uint224;
 
@@ -177,8 +177,8 @@
         uint amount1In = balance1 > _reserve1 - amount1Out ? balance1 - (_reserve1 - amount1Out) : 0;
         require(amount0In > 0 || amount1In > 0, 'UniswapV2: INSUFFICIENT_INPUT_AMOUNT');
         { // scope for reserve{0,1}Adjusted, avoids stack too deep errors
-        uint balance0Adjusted = balance0.mul(1000).sub(amount0In.mul(3));
-        uint balance1Adjusted = balance1.mul(1000).sub(amount1In.mul(3));
+        uint balance0Adjusted = balance0.mul(1000);
+        uint balance1Adjusted = balance1.mul(1000);
         require(balance0Adjusted.mul(balance1Adjusted) >= uint(_reserve0).mul(_reserve1).mul(1000**2), 'UniswapV2: K');
         }
 
